aiohttp/signals.py

```diff
@@ -0,0 +1,29 @@
+import asyncio
+
+
+class Signal(list):
+    """
+    Coroutine-based signal implementation
+
+    To connect a callback to a signal, use any list method. If wish to pass
+    additional arguments to your callback, use :meth:`functools.partial`.
+
+    Signals are fired using the :meth:`send` coroutine, which takes named
+    arguments.
+    """
+
+    @asyncio.coroutine
+    def send(self, **kwargs):
+        """
+        Sends data to all registered receivers.
+        """
+        for receiver in self:
+            res = receiver(**kwargs)
```

Comment:
Is it possible to ensure that all the receivers in Signal are callable? Otherwise it won't be cool to get some `TypeError: 'int' object is not callable` from the deepest aiohttp internals without any pointers about what the signal that was and where it came from.